Parsers/ASimNetworkSession/ProductParsers/NetworkSessionMicrosoftWindowsFirawall.yaml

```diff
@@ -0,0 +1,116 @@
+Parser:
+  Title: Microsoft Windows Firewall Network Sessions
+  Version: '0.1'
+  LastUpdated: Sep 1, 2021
+Product:
+  Name: Microsoft Windows Firewall
+Normalization:
+  Schema: NetworkSessions
+  Version: '0.2.0'
+References:
+- Title: ASIM Network Session Schema
+  Link: https://aka.ms/AzSentinelNetworkSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AzSentinelNormalization
+Description: |
+        This Query Parser maps Microsoft Windows Firewall Events (WindowsEvent) to the Azure Sentinel Information Model Network Session schema.
+        Event IDs which are parsed as part of this parser: 5150, 5151, 5152, 5153, 5154, 5155, 5156, 5167, 5158, 5159
+ParserName: vimNetworkSessionMicrosoftWindowsFirewall
+ParserQuery: |
+      let WindowsFirewall=(){
+      let LayerCodeTable = datatable (LayerCode:string,LayerName:string)[
+              '%%14596', 'IP Packet',
+              '%%14597', 'Transport',
+              '%%14598', 'Forward',
+              '%%14599', 'Stream',
+              '%%14600', 'Datagram Data',
+              '%%14601', 'ICMP Error',
+              '%%14602', 'MAC 802.3',
+              '%%14603', 'MAC Native',
+              '%%14604', 'vSwitch',
+              '%%14608', 'Resource Assignment',
+              '%%14609', 'Listen',
+              '%%14610', 'Receive/Accept',
+              '%%14611', 'Connect',
+              '%%14612', 'Flow Established',
+              '%%14614', 'Resource Release',
+              '%%14615', 'Endpoint Closure',
+              '%%14616', 'Connect Redirect',
+              '%%14617', 'Bind Redirect',
+              '%%14624', 'Stream Packet'];
+      let ProtocolTable = datatable (Protocol:int, NetworkProtocol: string)[
+              1, 'ICMP',
+              3, 'GGP',
+              6, 'TCP',
+              8, 'EGP',
+              12, 'PUP',
+              17, 'UDP',
+              20, 'HMP',
+              27, 'RDP',
+              46, 'RSVP',
+              47, 'PPTP data over GRE',
+              50, 'ESP',
+              51, 'AH',
+              66, 'RVD',
+              88, 'IGMP',
+              89, 'OSPF'];
+      let Directions = datatable (DirectionCode:string,NetworkDirection:string, isOutBound:bool)[
+                      '%%14592', 'Inbound', false,
+                      '%%14593', 'Outbound', true,
+                      '%%14594', 'Forward',false,
+                      '%%14595', 'Bidirectional', false];
+      let RawFireWallEvents = 
+          WindowsEvent
+          | where EventID between (5150 .. 5159)
+          | extend 
+              EventType = "NetworkSession",
```

Comment:
Push the constant lines - well any shared assignment for that matter - to after the union. Otherwise they are done twice as those lines precede the test for direction.